### PR TITLE
[PC-6846] Connect user to batch on app opening

### PR DESCRIPTION
**Pull request by viconnex**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-XXX

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: .env.production [hunk 1]
```diff
@@ -39,6 +39,10 @@ ID_CHECK_URL=https://id-check-front.passculture.app
 # Cultural Survey Typeform
 CULTURAL_SURVEY_TYPEFORM_URL=https://passculture.typeform.com/to/HGBAFB
 
+# Batch
```
Comment (blixit) [inline .env.production:42]: tu as mis à jour 1Password ?

File: src/api/helpers.ts [hunk 2]
```diff
@@ -71,9 +82,11 @@ export const safeFetch = async (
...
   try {
-    const tokenContent = jwtDecode(token) as JWTToken
+    // TODO: do not call jwtDecode when the token is undefined - modify jest setup so that jwtDecode is however called and returns a value
+    const tokenContent = jwtDecode<AccessToken>(token)
 
     if (tokenContent.exp * 1000 <= new Date().getTime()) {
+      // TODO: do not throw an error
       throw new Error('Token expired')
     }
   } catch (error) {
```
Comment (blixit) [inline src/api/helpers.ts:92]: je prépare une PR pour implémenter ce dont on a discuté: réutilisation de la fonction decodeAccessToken et utilisation de branchement au lieu de try-catch ?

File: src/features/auth/AuthContext.tsx [hunk 3]
```diff
@@ -26,6 +35,10 @@ export const AuthWrapper = ({ children }: { children: Element }) => {
   useEffect(() => {
     storage.readString('access_token').then((accessToken) => {
       setIsLoggedIn(!!accessToken)
+
+      if (accessToken) {
+        connectUserToBatchWithAccessTokenId(accessToken)
```
Comment (blixit) [inline src/features/auth/AuthContext.tsx:40]: il faut aussi le faire dans la routine de login

File: src/features/cheatcodes/pages/CheatCodes.tsx [hunk 4]
```diff
@@ -32,13 +33,28 @@ const oldAccesstoken =
...
   const [userEmail, setUserEmail] = useState('')
+  const [userId, setUserId] = useState<null | number>(null)
+
   useEffect(() => {
     getBatchInstallationID().then(setBatchInstallationId)
-    BatchUser.editor().setIdentifier('alainn').save()
   }, [])
+
+  useEffect(() => {
+    getUserId().then((userId) => setUserId(userId))
```
Comment (blixit) [inline src/features/cheatcodes/pages/CheatCodes.tsx:55]: en plus court et non moins lisible:
```ts
getUserId().then(setUserId)
```

File: .env.production (hunk 1 above)
Comment (viconnex) [reply .env.production:42]: les clés ne sont pas sur 1Password mais sur le dashboard de batch

File: src/features/auth/AuthContext.tsx (hunk 3 above)
Comment (viconnex) [reply src/features/auth/AuthContext.tsx:40]: c'est dans un autre ticket

Comment (viconnex) [reply src/features/auth/AuthContext.tsx:40]: ce sera fait dans un autre ticket https://passculture.atlassian.net/browse/PC-6847

**Review by AnoukHello — APPROVED** [review]

**Pull request merged**